### ISSUE TITLE
Solution to problem 29 does not pass

### ISSUE THREAD

**Issue by huylv** [oxalorg/4ever-clojure#59]
Here is my solution to [problem 29 -  Get the Caps](https://4clojure.oxal.org/#/problem/29):

```clojure
(fn [s] (apply str (filter #(<= (int \A) (int %) (int \Z)) s)))
```

Although it is even listed in the Solution page, it does not pass any test. Is this a bug?

**Comment by borkdude** [comment]
This is due to a difference in platform. Code in 4ever-clojure runs in JS and it behaves differently than the JVM:

```
user=> (int \A)
0
user=> (int \a)
0
```

The way you can do it in JS:

```
(.charCodeAt \a 0) ;;=> 97
```

@oxalorg We could patch SCI's `int` when invoked on a string to adapt to this, but I'm not sure it's a good idea.

**Comment by oxalorg** [comment]
Yes I think changing the behavior of `int` doesn't feel right. Then we're neither completely true to  Clojure nor completely true to Clojurescript and we're somewhere in between which makes me a bit uncomfortable.

I think it was a little different with `format` because goog.format is widely available and used, and we just aliased it to core.format.

I think it's fine that 4ever clojure is based on ClojureScript and somethings will fail from previous solutions. I'd rather just have a warning or a link to cheatsheet for clj->cljs.

**Comment by borkdude** [comment]
Agreed.
